core/src/main/java/org/apache/calcite/rel/core/Filter.java

```diff
@@ -110,6 +115,29 @@ public RexNode getCondition() {
     return condition;
   }
 
+  /**
+   * Check if any of the operands of the filter contains a
+   * correlation variable
+   */
+  public boolean hasCorrelation() {
+    if (condition instanceof RexCall) {
+      try {
+        RexVisitor<Void> visitor =
+            new RexVisitorImpl<Void>(true) {
+              public Void visitCorrelVariable(RexCorrelVariable var) {
+                throw new Util.FoundOne(var);
```

Comment:
Could you please use static exception instance, and, probably, static visitor instance?